### PR TITLE
Duplo 17696 plan waf

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added resource `duplocloud_plan_waf`
## Summary of changes

This PR does the following:

- Added resource `duplocloud_plan_waf`

- ...

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-17696 [TF]: Implement plan waf resource](https://app.clickup.com/t/8655600/DUPLO-17696)

**Review by tahir-duplo — COMMENTED** [review]
@Duplo-nikv Can you please add example usage and import as well.

File: duplocloud/resource_duplo_plan_waf.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		WebAclName:   d.Get("waf_name").(string),
+		WebAclId:     d.Get("waf_arn").(string),
+		DashboardUrl: d.Get("dashboard_url").(string),
+	}
+
+	err := c.PlanWAF(planID, &rq)
+	if err != nil {
+		return diag.Errorf("Error creating plan wafs for '%s': %s", planID, err)
+	}
+	d.SetId(planID)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_plan_waf.go:93]: It is better to set `plan-id/waf-name` for uniqueness of the resource

File: duplocloud/resource_duplo_plan_waf.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"waf_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"waf_arn": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_plan_waf.go:44]: Please add computed as well if it optional.

File: duplocloud/resource_duplo_plan_waf.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_plan_waf.go]: done

File: duplocloud/resource_duplo_plan_waf.go (hunk 2 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_plan_waf.go:44]: done

File: duplocloud/resource_duplo_plan_waf.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourcePlanWafRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	// Parse the identifying attributes
+	planID := d.Id()
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_plan_waf.go:60]: Split `id` and use plan id and waf name

File: duplocloud/resource_duplo_plan_waf.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+	diags := resourcePlanConfigsRead(ctx, d, m)
+	log.Printf("[TRACE] resourcePlanWafCreateOrUpdate(%s): end", planID)
+	return diags
+}
+
+func resourcePlanWafDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	// Parse the identifying attributes
+	planID := d.Id()
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_plan_waf.go:105]: Split `id` and use plan id and waf name

File: examples/resources/duplocloud_plan_images/import.sh [hunk 5]
```diff
@@ -1 +1 @@
-terraform import duplocloud_plan_images.myplan PLAN_ID
+terraform import duplocloud_plan_images.myplan PLAN_ID/WAF_NAME
```
Comment (tahir-duplo) [inline examples/resources/duplocloud_plan_images/import.sh:1]: Add Documentation for PLAN_ID and WAF_NAME

File: duplocloud/resource_duplo_plan_waf.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"waf_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"dashboard_url": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_plan_waf.go:49]: Please check if this is required.

File: duplocloud/resource_duplo_plan_waf.go [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"waf_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"waf_arn": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_plan_waf.go:44]: `waf_arn` must be required.

File: duplocloud/resource_duplo_plan_waf.go (hunk 3 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_plan_waf.go]: done

File: examples/resources/duplocloud_plan_images/import.sh (hunk 5 above)
Comment (Duplo-nikv) [reply examples/resources/duplocloud_plan_images/import.sh:1]: done

File: duplocloud/resource_duplo_plan_waf.go (hunk 4 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_plan_waf.go]: done

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**